File: scala/src/main/scala/tds/td3/common/io/EMailService.scala
package tds.td3.common.io

import tds.td3.common.mail.EMail

/**
  * A service that can send e-mails.
  * NON FUNCTIONAL: IO.
  */
trait EMailService:

  /**
    * Sends an e-mail.
    * NON FUNCTIONAL: IO.
    *
    * @param email the e-mail to be sent.
    */
  def send(email: EMail): Unit

File: java/app/src/main/java/org/example/tds/td3/promowarn/common/io/EMailService.java
package org.example.tds.td3.promowarn.common.io;

import org.example.tds.td3.promowarn.common.mail.EMail;

/**
 * An EMailService is anything that can send an EMail. That is, that has a
 * <code>send : EMail -> ()</code> operation. Examples are: a real email sending service,
 * or a logger for demonstration purposes.
 */
public interface EMailService {
    void send(final EMail email);
}
